CmakeLists: fix install libdir bindir

samuelig opened the pull request:
Previously it was fixed to <path>/lib but some distros install 64-bit
libraries in <path>/lib64.

Fixed bindir as well.

Signed-off-by: Samuel Iglesias Gonsálvez <siglesias@igalia.com>

samuelig committed:
CmakeLists: fix install libdir bindir

Previously it was fixed to <path>/lib but some distros install 64-bit
libraries in <path>/lib64.

Fixed bindir as well.

Signed-off-by: Samuel Iglesias Gonsálvez <siglesias@igalia.com>
Reviewed-by: Neil Roberts <nroberts@igalia.com>

bpeel commented:
Thanks for the patch.

It looks like it might need a bit more tweaking to work. When I try installing a build it ends up trying to install to `/pkg-config`, I guess because `$CMAKE_INSTALL_LIBDIR` isn’t defined. It looks like you need to do this in order to use it:

    include(GNUInstallDirs)

VkRunner is currently set to a minimum version of 2.82 of cmake, but that module seems to have been added in 3.0. I think we could safely changed the requirement to 3.0 because VK-GL-CTS also requires that now.

Next we seem to get a warning about not defining a language if we do that so I think we have to add this to the `project` define:

    project(VkRunner LANGUAGES C)

I think we’d have to fix the .pc file too because it’s currently hardcoded to `lib`.

samuelig commented:
Thanks for the review! I'll update the patch.

samuelig commented:
> Thanks for the patch.
> 
> It looks like it might need a bit more tweaking to work. When I try installing a build it ends up trying to install to `/pkg-config`, I guess because `$CMAKE_INSTALL_LIBDIR` isn’t defined. It looks like you need to do this in order to use it:
> 
> ```
> include(GNUInstallDirs)
> ```
> 
> VkRunner is currently set to a minimum version of 2.82 of cmake, but that module seems to have been added in 3.0. I think we could safely changed the requirement to 3.0 because VK-GL-CTS also requires that now.
> 

It was added in 2.8.5. However, to keep consistency to VK-GL-CTS I will updated it to 3.0.

> Next we seem to get a warning about not defining a language if we do that so I think we have to add this to the `project` define:
> 
> ```
> project(VkRunner LANGUAGES C)
> ```
> 
> I think we’d have to fix the .pc file too because it’s currently hardcoded to `lib`.

samuelig committed:
vkrunner.pc.in: pick libdir from cmake

Reviewed-by: Neil Roberts <nroberts@igalia.com>

samuelig committed:
CMakeLists: bump cmake version requirement to 3.0.2

Signed-off-by: Samuel Iglesias Gonsálvez <siglesias@igalia.com>
Reviewed-by: Neil Roberts <nroberts@igalia.com>

bpeel committed:
test-build.sh: Search for the pkgconfig path

Depending on the host system, the pkgconfig file could be installed in
lib or lib64. In order to make the test script work in either case, it
now just searches for the pkgconfig directory within the install prefix.

Reviewed-by: Samuel Iglesias Gonsálvez <siglesias@igalia.com>

bpeel commented:
Thanks for the updates. It looks good to me.

    Reviewed-by: Neil Roberts <nroberts@igalia.com>

I added an extra patch to your branch to fix the `test-build.sh` script. If you think that looks ok please push it along with your patches.
